### PR TITLE
Fix camera sync

### PR DESCRIPTION
**Pull request by hect95**
Configure cameras to run in `trigger_mode` to support external hardware syncronisation. Flir camera driver will not publish until cameras are triggered. This PR also:
 
* Adds pre-commit support
* Adds cyclone DDS as ROS RMW  + configurate it to support high msg throughput
* Avoid `dev.sh` override `latest` docker tag for convenience
* Enables colorised ROS log
* Add docker build and push workflow
  - `av_camera` is used as IMAGE_NAME rather than repository name
 * Simplify av_camera_launch pkg
   - Remove dev/test old launch files
   - Modify `chameleon_config.yaml` to run cameras in
      synchronised mode and explicitly use it when
      launching cameras
  * Synchronise host time with docker container

### CODE REVIEW

File: av_camera_launch/launch/all_cams.launch.xml [hunk 1]
```diff
@@ -1,58 +1,56 @@
...
-				<param name="camera_name" value="rsp_l" />
-				<param name="camera_type" value="chameleon" />
-				<param name="serial_number" value="'17496130'"/>
-				<param name="parameter_file" value="$(find-pkg-share spinnaker_camera_driver)/config/chameleon.yaml"/>
-				<param from="$(find-pkg-share av_camera_launch)/config/chameleon_config.yaml"/>
-			</composable_node>
-
-		</node_container>
-	</group>
+  <arg name="cfg_tpye" default="chameleon_sync"/>
```
Comment (GreatAlexander) [inline av_camera_launch/launch/all_cams.launch.xml:2]: "cfg_type"

**Review by GreatAlexander — COMMENTED** [review]
Should we simplify this pkg? We don't need so many launch files (those were created for testing), also shouldn't the synch config be the only config? Are we ever going to run the cameras unsynchronised or without a constant limited framerate?

**Comment by hect95** [comment]
> Should we simplify this pkg? We don't need so many launch files (those were created for testing), also shouldn't the synch config be the only config? Are we ever going to run the cameras unsynchronised or without a constant limited framerate?

Mmm I think is a good idea to simplify, I do not think we will ever run the cameras in unsynch mode. We can remove all of the testing launch files and leave `all_cams.launch.xml` only. Following the same idea, we can also remove the old parameters.

File: av_camera_launch/launch/all_cams.launch.xml (hunk 1 above)
Comment (hect95) [reply av_camera_launch/launch/all_cams.launch.xml]: Removing this argument as there is only one YAML param file :)

**Pull request merged**